tools/Google.Cloud.Tools.ReleaseManager/SuggestSmokeTestsCommand.cs

```diff
@@ -0,0 +1,164 @@
+            // Check that FooClient has FooSettings
+            .Where(t => assembly.GetType(t.FullName[..^6] + "Settings") is object)
+            .ToList();
+
+        private IEnumerable<SmokeTest> SuggestSmokeTests(Type client) =>
+            client.GetMethods(BindingFlags.Public | BindingFlags.Instance)
+                // All RPC methods have a final CallSettings parameter
+                .Where(method => method.GetParameters().LastOrDefault()?.ParameterType.FullName == "Google.Api.Gax.Grpc.CallSettings")
+                // Filter out async methods.
+                .Where(method => !typeof(Task).IsAssignableFrom(method.ReturnType))
+                .Where(method => !method.ReturnType.FullName.StartsWith("Google.Api.Gax.PagedAsyncEnumerable"))
+                // Now collect into overloaded methods
+                .GroupBy(method => method.Name)
+                .Select(SyncRpc.FromMethods)
+                // And convert to smoke tests, where possible
+                .Select(rpc => rpc.ToSmokeTestOrNull())
+                .Where(test => test is object);
+
+        private class SyncRpc
+        {
+            public string Client { get; }
+            public string Method { get; }
+            public IReadOnlyList<Overload> Overloads { get; }
+
+            private SyncRpc(string client, string method, IReadOnlyList<Overload> overloads) =>
+                (Client, Method, Overloads) = (client, method, overloads);
+
+            public static SyncRpc FromMethods(IEnumerable<MethodInfo> methods) =>
+                new SyncRpc(methods.First().DeclaringType.Name, methods.First().Name, methods.Select(Overload.FromMethod).ToList());
+
+            public SmokeTest ToSmokeTestOrNull()
+            {
+                // We just need to fill in the arguments...
+                var candidate = new SmokeTest { Client = Client, Method = Method };
+
+                // Parameterless RPCs are very rare, but are good candidates for smoke tests.
+                // Admittedly this has some false positives, e.g. in Vision, ProductSearchClient.PurgeProducts.
+                if (Overloads.Any(o => o.RequiredParameters.Count == 0))
+                {
+                    // No arguments required
+                    return candidate;
+                }
+                // If there's a method that accepts just a ProjectName, find its equivalent accepting a string. (There should be exactly one.)
+                if (Overloads.Any(o => o.HasSingleParameterOfType("Google.Api.Gax.ResourceNames.ProjectName")))
+                {
+                    var stringParameter = Overloads.Single(o => o.HasSingleParameterOfType(typeof(string))).RequiredParameters[0];
+                    candidate.Arguments[stringParameter.Name] = "projects/${PROJECT_ID}";
+                    return candidate;
+                }
+
+                // If there's a method that accepts just a LocationName, find its equivalent accepting a string. (There should be exactly one.)
```

Comment:
Serious errors here, which need fixing before any further progress can be made in this PR.
There should not be a full stop before the open parenthesis, and the first character in the parenthesis should not be capitalized.  The full stop within the parenthesis should then be moved out of the parenthesis.
Alternatively consider not using parenthesis at all; I enjoy using a semicolon for sentence structures such as this - or possibly a hyphen.

Reply:
https://www.youtube.com/watch?v=kfVsfOSbJY0